src/gui/plugins/joint_position_controller/Joint.qml

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+import QtQuick 2.9
+import QtQuick.Controls 2.2
+import QtQuick.Layouts 1.3
+import "qrc:/JointPositionController"
+import "qrc:/qml"
+
+Rectangle {
+  id: joint
+  height: slider.height
+  width: jointPositionController.width
+  color: index % 2 == 0 ? lightGrey : darkGrey
+
+  // Position value
+  property double value: 0.0
+
+  // Horizontal margins
+  property int margin: 15
+
+  Connections {
+    target: joint
+    onValueChanged: {
+      jointPositionController.onCommand(model.name, joint.value);
+      spin.value = joint.value;
+      slider.value = joint.value;
+    }
+  }
+
+  RowLayout {
+    anchors.fill: parent
+    spacing: 0
+
+    Item {
+      height: parent.height
+      width: margin
+    }
+
+    Text {
+      text: model.name
+      Layout.alignment: Qt.AlignVCenter
+      Layout.preferredWidth: 100
+      elide: Text.ElideRight
+      ToolTip {
+        visible: ma.containsMouse
+        delay: Qt.styleHints.mousePressAndHoldInterval
+        text: model.name
+        y: -30
+        enter: null
+        exit: null
+      }
+      MouseArea {
+        id: ma
+        anchors.fill: parent
+        hoverEnabled: true
+        acceptedButtons: Qt.RightButton
+      }
+    }
+
+    IgnSpinBox {
+      id: spin
+      value: model.value
+      minimumValue: model.min
+      maximumValue: model.max
+      decimals: 2
+      onEditingFinished: {
+        joint.value = spin.value
+      }
```

Comment:
```suggestion
      stepSize: 0.1
      onValueChanged: {
        joint.value = spin.value
      }
```
Currently, the `stepSize` is `1` but I think `0.1` would be a better increment. Also, I think using `onValueChanged` instead of `onEditingFinished` is more user friendly, the user doesn't have to hit enter every time they want to update a joint when using the spinbox

Reply:
Both are very good suggestions. I started with both of them, but I had to revert to using `onEditingFinished` when I worked on the reset button.

While working on reset, I noticed that the values on the GUI were not being updated to reflect the actual state of the robot. When I fixed that, I realized I couldn't use `onValueChanged`, because that signal doesn't differentiate the value being changed by the user, or from the ECM. So what was happening was that the intermediate state of a joint moving would be sent as a new command, and the joint would tend to return to its original position in a negative feedback loop. It's hard to explain with words, see what happened with `onValueChanged`:

![jointfeedback](https://user-images.githubusercontent.com/5751272/111403583-77e46e00-868a-11eb-9915-795c52aff45b.gif)


This is all on 08904d0, let me know what you think.